### PR TITLE
[C-2807] Add ModalField subforms with cancellation

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Adds a subform layer to the upload form popup modals so the changes can be cancelled or accepted. Values in the upload form won't be updated until the modal form is actively submitted by the user. If they click out or close the modal without submission, the values will revert.

TODO:
- On cancel, values currently revert to their original default value, not to the last saved value

### Dragons

We're adding a layer of complexity. Caution when calling `useField` as it refers to the closest parent `<Formik>`'s context.

### How Has This Been Tested?

Local web

### Feature Flags ###

`UPLOAD_REDESIGN_ENABLED`

### CODE REVIEW

File: packages/web/src/pages/upload-page/components/TrackModalArray.tsx [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
 export const TrackModalArray = () => {
+  const { values: currentValues, setValues } =
+    useFormikContext<EditFormValues>()
+
+  const handleSubmit = (values: Partial<EditFormValues>) => {
+    // TODO: reformat values to match outer form structure
+    setValues({ ...currentValues, ...values })
+  }
+
+  const [{ value: releaseDateValue }] = useField(RELEASE_DATE_FIELD_NAME)
```
[the remark this answers is not in the document]
Comment (dylanjeffers) [reply packages/web/src/pages/upload-page/components/TrackModalArray.tsx:26]: So it establishes the field name for the outer field, the handle submit logic, and setting up the inner form?

File: packages/web/src/pages/upload-page/fields/ReleaseDateModalForm.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        preview={preview}
+      >
+        <h3 className={cn(styles.title, styles.modalHeading)}>
+          {messages.title}
+        </h3>
+        <p className={styles.description}>{messages.description}</p>
+        <div className={styles.datePicker}>
+          <DatePickerField
+            name={RELEASE_DATE_FIELD_NAME}
+            label={'Release Date'}
```
Comment (dylanjeffers) [inline packages/web/src/pages/upload-page/fields/ReleaseDateModalForm.tsx:71]: Messages obj

**Review by dylanjeffers — APPROVED** [review]
This looks so money. Lmk what you think about defining the useField and handlesubmit for each modal field

File: packages/web/src/pages/upload-page/components/TrackModalArray.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/web/src/pages/upload-page/components/TrackModalArray.tsx:26]: Yes. Definitely agree

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-upload-subform

File: packages/web/src/pages/upload-page/components/TrackModalArray.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/web/src/pages/upload-page/components/TrackModalArray.tsx:26]: Actually, this is referring to the field in the outer form, so I'm wondering if it's wrong to encapsulate the knowledge of the outer form within the individual subforms. What do you think?

**Comment by amendelsohn** [comment]
image
Values now being correctly passed through to the end

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-upload-subform

**Pull request merged**